packages/smooth_app/lib/pages/product/common/product_app_bar.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_back_button.dart';
+import 'package:smooth_app/widgets/smooth_app_bar.dart';
+import 'package:smooth_app/widgets/smooth_floating_message.dart';
+
+class ProductAppBar extends StatefulWidget implements PreferredSizeWidget {
+  const ProductAppBar(
+      {required this.barcodeVisibleInAppbar,
+      required this.productName,
+      required this.productBrand,
+      required this.barcode});
+
+  final String productName;
+  final String productBrand;
+  final String barcode;
+  final bool barcodeVisibleInAppbar;
+
+  @override
+  Size get preferredSize => const Size.fromHeight(kToolbarHeight);
```

Comment:
Not sure about this one, as I couldn't find it in the original code.

Reply:
Since its implementing `PreferredSizeWidget` !
Initially it was directly in the `edit_product_page.dart`. Now its on seperate file ! and appBar specifically takes preferredsizewidget! so had to add that .